src/azure.rs

```diff
@@ -393,12 +392,49 @@ impl ObjectStore for MicrosoftAzure {
                 from: from.as_ref(),
                 to: to.as_ref(),
             })?;
+        Ok(())
+    }
 
+    async fn copy_if_not_exists(&self, from: &Path, to: &Path) -> Result<()> {
+        let from_url = self.get_from_url(from)?;
+        self.container_client
+            .as_blob_client(to.as_ref())
+            .copy(&from_url)
+            .if_match_condition(IfMatchCondition::NotMatch("*".to_string()))
+            .execute()
+            .await
+            .map_err(|err| {
+                print!("ERROR --- {:?}", err);
+                if let Some(azure_core::HttpError::StatusCode { status, .. }) =
+                    err.downcast_ref::<azure_core::HttpError>()
+                {
+                    if status.as_u16() == 409 {
+                        return Error::AlreadyExists {
+                            source: err,
+                            path: to.to_string(),
+                        };
+                    };
+                };
+                Error::UnableToCopyFile {
+                    source: err,
+                    container: self.container_name.clone(),
+                    from: from.to_string(),
+                    to: to.to_string(),
+                }
+            })?;
         Ok(())
     }
+}
 
-    async fn copy_if_not_exists(&self, _from: &Path, _to: &Path) -> Result<()> {
-        Err(crate::Error::NotImplemented)
+impl MicrosoftAzure {
+    fn get_from_url(&self, from: &Path) -> Result<reqwest::Url> {
```

Comment:
```suggestion
    fn get_copy_from_url(&self, from: &Path) -> Result<reqwest::Url> {
```

And perhaps a docstring, I initially thought this was a method for converting from a URL, and was then confused :laughing: